app/main/menu.js

```diff
@@ -162,6 +162,20 @@ class AppMenu {
 					ConfigUtil.setConfigItem('showSidebar', newValue);
 				}
 			}
+		}, {
+			label: 'Auto hide Menu bar',
+			checked: ConfigUtil.getConfigItem('autoHideMenubar'),
```

Comment:
@priyank-p, @abhigyank I have a doubt here. What would happen here if the config files fail on startup? Maybe we should use the following to avoid that situation -

```
checked: ConfigUtil.getConfigItem('autoHideMenubar') || false ,
```

Reply:
@akashnimare I believe I can answer that.

> What would happen here if the config files fail on startup?

This could mean two things, I think - either the `settings.json` file is completely malformed, in which case there'll be an error saying "There was an error in saving the settings" and the default settings will be set, or the field is absent and the `checked` field is therefore `null`. 
That would be the same as setting it to `false`, but I agree that my implementation could be better. 

I'll set it to:
```javascript
checked: ConfigUtil.getConfigItem('autoHideMenubar', false)
```

Reply:
I'll also add a default setting of `autoHideMenubar: false` in `main.js`.